pkgs/top-level/all-packages.nix

```diff
@@ -3692,6 +3692,8 @@ with pkgs;
 
   deterministic-uname = callPackage ../build-support/deterministic-uname { };
 
+  deterministic-host-uname = deterministic-uname.override { forPlatform = stdenv.hostPlatform; };
```

Comment:
When this is added to `nativeBuildInputs` it'll just be the `buildPlatform` uname so it's kinda useless
```
$ nix run ".#pkgsCross.aarch64-multiplatform.buildPackages.deterministic-host-uname" -- -a
Linux nixpkgs unknown #1-NixOS Tue Jan 1 00:00:00 UTC 1980 x86_64 GNU/Linux
```

Reply:
You're right, when I tested it the inputs weren't spliced :/ 

Though `nativeBuildInputs` is `pkgsBuildHost` so it should be aware of the host system that you're building for. Is it just an incorrect offset?

Reply:
Switching it to `targetPlatform` solves the issue. This should be fine given everything is offset by 1 right?

Reply:
Fixed in #341256 341256